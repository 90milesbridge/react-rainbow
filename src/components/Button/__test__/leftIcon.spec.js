import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import LeftIcon from './../leftIcon';

describe('<LeftIcon in the Button compoent/>', () => {
    it('should not have children when the iconName is not passed', () => {
        const component = mount(
            <LeftIcon position="left" />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should not have children when the position is other than left', () => {
        const component = mount(
            <LeftIcon iconName="utility:world" position="right" />,
        );
        expect(component.children().length).toBe(0);
    });
    it('should have the right class names when the iconName is passed and position is left', () => {
        const component = renderer.create(
           <LeftIcon iconName="utility:world" position="left" />,
       );
        expect(component).toMatchSnapshot();
    });
});
