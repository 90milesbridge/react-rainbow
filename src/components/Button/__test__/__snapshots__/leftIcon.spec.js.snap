// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeftIcon in the Button compoent/> should have the right class names when the iconName is passed and position is left 1`] = `
<svg
  aria-hidden={true}
  className="slds-button__icon slds-button__icon_left"
  style={Object {}}
>
  <use
    xlinkHref="/icons/utility-sprite/svg/symbols.svg#world"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  />
</svg>
`;
